tests/scripts/auto-grow-storage.sh

```diff
@@ -0,0 +1,210 @@
+   echo "Looping at $(date +"%Y-%m-%d %H:%M:%S")"
+
+   while true
+   do
+      if [ "${total_alerts}" == "" ]
+      then
+         echo "Alert manager not configured,re-run the script"
+         exit 1
+      fi
+      export entry
+      entry=$( jq ".data.alerts[$itr]" < tt.txt)
+      thename=$(echo "${entry}" | jq -r '.labels.alertname')
+      if [ "${thename}" = "CephOSDNearFull" ] || [ "${thename}" = "CephOSDCriticallyFull" ]
+      then
+         echo "${entry}"
+         ns=$(echo "${entry}" | jq -r '.labels.namespace')
+         osdID=$(echo "${entry}" | jq -r '.labels.ceph_daemon')
+         osdID=${osdID/./-}
+         pvc=$(oc get deployment -n rook-ceph rook-ceph-"${osdID}" -o json | jq -r '.metadata.labels."ceph.rook.io/pvc"')
+         echo "Processing NearFull or Full alert for PVC ${pvc} in namespace ${ns}"
+         if [[ $1 == "count" ]]
+         then 
+            growHorizontally "$2" "${pvc}" "${ns}" "$3"
+         else
+            growVertically "$2"  "${pvc}" "${ns}" "$3"
+         fi
+      fi   
+      (( itr = itr + 1 ))
+       if [[ "${itr}" == "${total_alerts}" ]] || [[ "${total_alerts}" == "0" ]]
+      then
+         sleep 600
+         rm -f ./tt.txt
+         curl  -s http://127.0.0.1:9090/api/v1/alerts | jq -r '.' >./tt.txt
+         total_alerts=$( jq '.data.Alerts | length' < tt.txt)
+         itr=0
+         echo "Looping at $(date +"%Y-%m-%d %H:%M:%S")"
+      fi  
+   done
+}
+
+function creatingPrerequisites(){
+   echo "creating Prerequisites deployments"
+   # creating Prometheus operator 
+   kubectl apply -f https://raw.githubusercontent.com/coreos/prometheus-operator/v0.40.0/bundle.yaml
+   # waitng for Prometheus operator to get ready
+   timeout 30 sh -c "until [ "$(kubectl get pod -l app.kubernetes.'io/name'=prometheus-operator -o json | jq -r '.items[0].status.phase')" = "Running" ]; do echo 'waiting for prometheus-operator to get created' && sleep 1; done"
+   # creating a service monitor that will watch the Rook cluster and collect metrics regularly
+   kubectl create -f https://raw.githubusercontent.com/rook/rook/master/cluster/examples/kubernetes/ceph/monitoring/service-monitor.yaml
+   # create the PrometheusRule for Rook alerts.
+   kubectl create -f https://raw.githubusercontent.com/rook/rook/master/cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml
+   # create prometheus-rook-prometheus-0 pod
+   kubectl create -f https://raw.githubusercontent.com/rook/rook/master/cluster/examples/kubernetes/ceph/monitoring/prometheus.yaml
+   # waitng for prometheus-rook-prometheus-0 pod to get ready
+   timeout 60 sh -c "until [ "$(kubectl get pod -l prometheus=rook-prometheus -nrook-ceph -o json | jq -r '.items[0].status.phase')" = "Running" ]; do echo 'waiting for prometheus-rook-prometheus-0 pod to get created' && sleep 1; done"
+   # port-forward to access the cluster outside in browser
+   kubectl port-forward service/prometheus-operated 9090 -nrook-ceph
```

Comment:
I want to use `prometheus-operated` service outside the cluster, for this currently, I am doing port-forwarding to access it in the browser.

I want to ask what would I use for production deployment.
I thought of using ingress for it but it would be needing an extra yaml configuration..

Reply:
As discussed, let's see if the script can be run inside the cluster so we can avoid this question of accessing prometheus from outside the cluster